msprime/mutations.py

```diff
@@ -132,15 +245,7 @@ def mutate(
     seed = random_seed
     if random_seed is None:
         seed = simulations._get_random_seed()
-    seed = int(seed)
```

Comment:
Reinstating this line fixes the first broken test.

Reply:
(I'm looking into the others, seems a low-level class can be passed as a param, which the provenance doesn't support)

Reply:
We can refactor around that - probably shouldn't be making low-level classes part of the public API anyway.

Reply:
Yep, it's `IntervalMap`. Just looking into if I should wrap it, or have it's arguments passed as params to InfiniteSites et al.